api/geth_backend.go

```diff
@@ -1492,6 +1492,10 @@ func (b *GethStatusBackend) prepareNodeAccount(request *requests.CreateAccount,
 		return nil, errors.Wrap(err, "failed to prepare settings")
 	}
 
+	if response.account.Name == "" {
+		response.account.Name = response.settings.Name
```

Comment:
Just double-checking @Parveshdhull, but are you sure `Name` is the only fallback value we should use? I ask this because in `status-mobile` we had many bugs because of misunderstandings about which name to pick and their order.

In `status-mobile` I think we are prioritizing the `DisplayName` over `Name` if the value isn't blank https://github.com/status-im/status-mobile/blob/b11d3bed780cfb7f3dea8bd7f7bd1780d15200dc/src/status_im/contexts/profile/utils.cljs#L16.

Also here https://github.com/status-im/status-mobile/blob/b11d3bed780cfb7f3dea8bd7f7bd1780d15200dc/src/status_im/subs/contact/utils.cljs#L69

Reply:
Thank you @ilmotta for reviewing the PR. You are right DisplayName takes precedence over Name.

DisplayName is the value provided by the user while creating an account. While the name is just 3 words random name generated from the public key.

https://github.com/status-im/status-go/blob/d794e433476db879f34893a61254bd501abdabb2/api/defaults.go#L67

So in this PR, we are making sure if a user creates a new account with an empty displayName (in case of syncing fallback), at least he/she will have 3 random words as fallback (not as a higher precedence)

Reply:
btw, It will be replaced with DisplayName as soon as syncing complete

Reply:
Thank you for the explanation. Approving